### PR TITLE
Fix XVR test exclusions

### LINKED ISSUE #30353: [Failing Test]: XVR tests running on VR test suites causing test failure
### What happened?

#29924 merged core-construction-java into java-core. After that many validation runner tests failed:

```
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$GroupByKeyTest
FAILED	0.076s	0.076s	0.076s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$SingleInputOutputTest
FAILED	0.071s	0.071s	0.071s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$FlattenTest
FAILED	0.066s	0.066s	0.066s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$PythonDependenciesTest
FAILED	0.066s	0.066s	0.066s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$MultiInputOutputWithSideInputTest
FAILED	0.065s	0.065s	0.065s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$CombineGloballyTest
FAILED	0.061s	0.061s	0.061s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$CoGroupByKeyTest
FAILED	0.056s	0.056s	0.056s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$CombinePerKeyTest
FAILED	0.056s	0.056s	0.056s	
org.apache.beam.sdk.util.construction.ValidateRunnerXlangTest$PartitionTest
FAILED	0.051s	0.051s	0.051s	
```
These tests are not supposed to run in VR test suites. On the other hand, also checked that XVR test suites no longer run these tests.

We should exclude these tests from running in VR while add them back to XVR

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Abacn**
fix #30353

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
- Comparing https://github.com/apache/beam/actions/runs/7977632225 and https://github.com/apache/beam/actions/runs/7978107238?pr=30354, 9 tests are executed in XVR PostCommit again

- VR PostCommit permared before now back green (other than flaky ones)

**Comment by Abacn** [comment]
R: @chamikaramj @kennknowles

**Comment by Abacn** [comment]
R: @damccorm

File: runners/google-cloud-dataflow-java/build.gradle [hunk 1]
```diff
@@ -192,6 +191,7 @@ def commonLegacyExcludeCategories = [
 ]
 
 def commonRunnerV2ExcludeCategories = [
+        'org.apache.beam.sdk.testing.UsesExternalService',
```
Comment (chamikaramj) [inline runners/google-cloud-dataflow-java/build.gradle:194]: Will this result in all Runner V2 test suites not running x-lang tests (essentially disabling all x-lang tests) ?

Comment (chamikaramj) [reply runners/google-cloud-dataflow-java/build.gradle:194]: Ah, thanks. We should fix the name in that case.

**Comment by chamikaramj** [comment]
LGTM (minus test failures)

**Comment by Abacn** [comment]
Failed tests:

- beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming
  - run1: org.apache.beam.sdk.transforms.CombineTest$CombineWithContextTests > testHotKeyCombineWithSideInputs
  - run2: 
- beam_PostCommit_Java_PVR_Spark3_Streaming
  - actually successed in 1h58min, time out on clean up, opened #30356 to bump timeout
- beam_PostCommit_Java_PVR_Spark_Batch
  - run1: testWindowingInStartAndFinishBundle (org.apache.beam.sdk.transforms.ParDoTest$LifecycleTests) 
  - run2: many tests failed pipeline file not found. This is likely a racing condition
  - actually successed in 1h58min, time out on clean up, opened #30356 to bump timeout
- beam_PostCommit_Java_ValidatesRunner_Flink_Java11
  - run1: org.apache.beam.sdk.transforms.ParDoLifecycleTest > testTeardownCalledAfterExceptionInStartBundleStateful
  - run2: org.apache.beam.sdk.transforms.ParDoLifecycleTest > testTeardownCalledAfterExceptionInProcessElementStateful

**Pull request merged**